core/src/main/java/com/redhat/lightblue/client/request/data/GenerateRequest.java

```diff
@@ -0,0 +1,98 @@
+package com.redhat.lightblue.client.request.data;
+
+
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.redhat.lightblue.client.Operation;
+import com.redhat.lightblue.client.Projection;
+import com.redhat.lightblue.client.Query;
+import com.redhat.lightblue.client.Sort;
+import com.redhat.lightblue.client.http.HttpMethod;
+import com.redhat.lightblue.client.request.AbstractLightblueDataRequest;
+
+/**
+ * Generate a value using a field's value generator
+ *
+ * path: Path for the field whose value generator will be invoked
+ * n: Number of values to generate, defaults to 1
+ *
+ * Parse the results as:
+ * <pre>
+ *	String[] results = client.data(request, String[].class);
+ * </pre>
+ * 
+ */
+public class GenerateRequest extends AbstractLightblueDataRequest {
+
+    private Integer n;
+    private String path;
```

Comment:
Only if I remove the path() function, and use this as an immutable class
with only ctors. But then i'll lose the ability to have ctors for default
versions, because that'll be ambiguous. Still, can be done. Should I?

That still doesn't prevent passing in null for path to ctor though.,

On Wed, May 4, 2016 at 1:26 PM, Naveen Malik <notifications@github.com>
wrote:

> In
> core/src/main/java/com/redhat/lightblue/client/request/data/GenerateRequest.java
> <https://github.com/lightblue-platform/lightblue-client/pull/292#discussion_r62099270>
> :
>
> > +/**
> > + * Generate a value using a field's value generator
> > + *
> > + * path: Path for the field whose value generator will be invoked
> > + * n: Number of values to generate, defaults to 1
> > + *
> > + * Parse the results as:
> > + * <pre>
> > + *	String[] results = client.data(request, String[].class);
> > + * </pre>
> > + *
> > + */
> > +public class GenerateRequest extends AbstractLightblueDataRequest {
> > +
> > +    private Integer n;
> > +    private String path;
>
> Is there a way to make path required? Clients could create this using
> constructor without path then try to make the request without calling
> path(...)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-client/pull/292/files/85e4994abeffb4aebf1f052b533209bcd6b6c1e1#r62099270>
>